### PR TITLE
Streamline progress printing in cmd/restic

### PR DESCRIPTION
**Pull request by greatroar**
What does this PR change? What problem does it solve?
-----------------------------------------------------

It changes the progress printing in cmd/restic, with the following main features:

* PrintProgress no longer does unnecessary Sprintf calls
* newProgressMax's callback checks whether the terminal supports line updates once instead of once per call
* the callback looks up the terminal width once per call instead of twice (on Windows)
* the status shortening now uses the Unicode-aware version from internal/ui/termstatus; it doesn't need this, so consider this code reuse and future-proofing
* ASCII handling in termstatus.Truncate has been optimized

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

No.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/ui/termstatus/status_test.go [hunk 1]
```diff
@@ -19,17 +19,40 @@ func TestTruncate(t *testing.T) {
 		{"foo", 0, ""},
 		{"foo", -1, ""},
 		{"Löwen", 4, "Löwe"},
-		{"あああああああああ/data", 10, "あああああ"},
-		{"あああああああああ/data", 11, "あああああ"},
```
Comment (MichaelEischer) [inline internal/ui/termstatus/status_test.go:23]: I think we should also keep this test, which checks that a unicode character is not split in half.

**Review by MichaelEischer — COMMENTED** [review]
The PR looks good. I only have a tiny nitpick regarding the test case.

File: internal/ui/termstatus/status_test.go (hunk 1 above)
Comment (greatroar) [reply internal/ui/termstatus/status_test.go:23]: Good point. Restored a version of it.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks for the cleanup :-)
